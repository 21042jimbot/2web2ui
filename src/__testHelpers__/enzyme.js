export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
