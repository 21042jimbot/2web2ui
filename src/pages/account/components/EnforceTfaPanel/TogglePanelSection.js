import React from 'react';
import styled from 'styled-components';
import {
  Banner,
  Panel,
  Stack,
  Toggle,
  Columns,
  Column,
  ScreenReaderOnly,
} from 'src/components/matchbox';
import LabelledValue from 'src/components/labelledValue/LabelledValue';

const RightAlignedText = styled.div`
  text-align: right;
`;

export const TogglePanelSection = ({
  readOnly,
  tfaRequired,
  toggleTfaRequired,
  tfaRequiredEnforced,
}) => {
  const tfaRequiredOrEnforced = tfaRequired || tfaRequiredEnforced;
  const tfaRequiredMsg = tfaRequiredOrEnforced
    ? 'All users must have two-factor authentication enabled to login to this account.'
    : 'Each user can manage their own two-factor authentication settings.';
  return (
    <Panel.LEGACY.Section>
      <Stack>
        <Columns>
          <Column width={2 / 3}>
            {tfaRequiredEnforced && (
              <Banner size="small">Two-factor authentication is required for your account.</Banner>
            )}
          </Column>
        </Columns>
        <Columns>
          <Column width={2 / 3}>
            <LabelledValue label="Status">
              <strong>{tfaRequiredOrEnforced ? 'Required' : 'Optional'}</strong>
              <p>{tfaRequiredMsg}</p>
            </LabelledValue>
          </Column>
          <Column width={1 / 3}>
            <RightAlignedText>
              <Toggle
                label={<ScreenReaderOnly>Enforce TFA</ScreenReaderOnly>}
                labelHidden
                disabled={readOnly || tfaRequiredEnforced}
                checked={tfaRequiredOrEnforced}
                onChange={toggleTfaRequired}
              />
            </RightAlignedText>
          </Column>
        </Columns>
      </Stack>
    </Panel.LEGACY.Section>
  );
};

export default TogglePanelSection;
