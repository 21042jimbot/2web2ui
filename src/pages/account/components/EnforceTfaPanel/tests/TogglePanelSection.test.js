import React from 'react';
import { shallow } from 'enzyme';
import TogglePanelSection from '../TogglePanelSection';

describe('Component: TogglePanel', () => {
  const baseProps = {
    tfaRequired: false,
    toggleTfaRequired: () => {},
  };

  function subject(props) {
    return shallow(<TogglePanelSection {...baseProps} {...props} />);
  }

  it('should reflect tfaRequired', () => {
    expect(
      subject({ tfaRequired: true })
        .find('Toggle')
        .prop('checked'),
    ).toEqual(true);
  });

  it('should call back on toggle', () => {
    const toggleTfaRequired = jest.fn();
    const wrapper = subject({ toggleTfaRequired });
    wrapper.find('Toggle').simulate('change');
    expect(toggleTfaRequired).toHaveBeenCalledTimes(1);
  });

  it('should be disabled if readOnly or tfaEnforced is set to true', () => {
    expect(
      subject({ readOnly: true })
        .find('Toggle')
        .prop('disabled'),
    ).toEqual(true);
    expect(
      subject({ tfaRequiredEnforced: true })
        .find('Toggle')
        .prop('disabled'),
    ).toEqual(true);
  });
});
