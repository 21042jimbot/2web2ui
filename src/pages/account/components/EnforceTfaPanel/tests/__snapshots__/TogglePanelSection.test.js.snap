// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TogglePanel should render 1`] = `
<Panel.Section>
  <Grid>
    <Grid.Column
      md={11}
      xs={12}
    >
      <LabelledValue
        label="Status"
      >
        <strong>
          Optional
        </strong>
        <p>
          Each user can manage their own two-factor authentication settings.
        </p>
      </LabelledValue>
    </Grid.Column>
    <Grid.Column
      md={1}
      style={
        Object {
          "textAlign": "right",
        }
      }
      xs={12}
    >
      <Toggle
        checked={false}
        id="enforceTfa"
        onChange={[Function]}
      />
    </Grid.Column>
  </Grid>
</Panel.Section>
`;
